packages/ui/src/ConditionalWrap.tsx

```diff
@@ -0,0 +1,60 @@
+import { ReactNode } from 'react';
+
+export interface ConditionalWrapProps {
+  if: boolean;
+  then: (children: ReactNode) => ReactNode;
+  else?: (children: ReactNode) => ReactNode;
+  children: ReactNode;
+}
+
+/**
+ * Internal utility component to optionally wrap a React component with another
+ * React component, depending on a condition.
+ *
+ * @example
+ * ```tsx
+ * <ConditionalWrap
+ *   if={shouldUseTooltip}
+ *   then={(children) => (
+ *     <Tooltip>
+ *       <TooltipTrigger>{children}</TooltipTrigger>
+ *       <TooltipContent>Here is the tooltip text.</TooltipContent>
+ *     </Tooltip>
+ *   )}
+ * >
+ *   Here is the content that may or may not need a tooltip.
+ * </ConditionalWrap>
+ * ```
+ *
+ * You can also pass an `else` prop to wrap the `children` if the condition is
+ * _not_ met.
+ *
+ * @example
+ * ```tsx
+ * <ConditionalWrap
+ *   if={shouldUseTooltip}
+ *   then={(children) => (
+ *     <Tooltip>
+ *       <TooltipTrigger>{children}</TooltipTrigger>
+ *       <TooltipContent>Here is the tooltip text.</TooltipContent>
+ *     </Tooltip>
+ *   )}
+ *   else={(children) => (
+ *     <NonTooltipWrapper>{children}</NonTooltipWrapper>
+ *   )}
+ * >
+ *   Here is the content that may or may not need a tooltip.
+ * </ConditionalWrap>
+ * ```
+ *
+ * @see https://stackoverflow.com/a/56870316/974981
+ */
+export const ConditionalWrap = ({
```

Comment:
suggestion: could you move the file somewhere from the `src`? I feel like this directory should only have directories inside. As for shared components or utils, they could go into `shared` or `utils` without `index.tsx` in them, so it won't be exported outside the library

Reply:
Done! 

FWIW, components in src will not get exported if they're not structured like `ComponentName/index.tsx` (see our [`exports`](https://github.com/penumbra-zone/web/blob/main/packages/ui/package.json#L24)).

That said, I agree with your suggestion to move it to avoid confusion.